common/FileUtils.java

```diff
@@ -472,5 +478,65 @@ public Message readPrototxtFromZipUnchecked(
   public File getFile(String path) {
     return new File(expandHomeDirectory(path));
   }
+
+  /**
+   * Gets file names in path by pattern. The pattern can containing the * wildcard: this wildcard
+   * matches any string excluding the directory separator /. In addition, filename patterns can
+   * contain the recursive ** wildcard. This wildcard will match zero or more complete path segments
+   * separated by the directory separator /. Throws IllegalStateException if path is not a folder.
+   */
+  public ImmutableList<String> findFilesByGlobPattern(String path, String pattern)
+      throws IOException {
+    if (!folderExists(path)) {
+      throw new IllegalStateException(String.format("%s folder does not exist", path));
+    }
+    Set<String> result = new HashSet<>();
+    final String filenamePattern = pattern.substring(pattern.lastIndexOf('/') + 1);
+    final PathMatcher matcher = fileSystem.getPathMatcher("glob:" + filenamePattern);
```

Comment:
I'm looking at the options for glob here:
https://docs.oracle.com/javase/8/docs/api/java/nio/file/FileSystem.html#getPathMatcher-java.lang.String-

Can you explain the case that is not covered there, for which you need to write custom code? Specifically, I mean not using whole `pattern` and only `filenamePattern`, and also needing `areFoldersMatchingPattern`.